server/handler.go

```diff
@@ -321,7 +321,7 @@ func (h *Handler) doQuery(
 	}()
 	if err != nil {
 		logrus.Tracef("Error running query %s: %s", query, err)
-		return err
+		return sql.CastSQLError(err)
```

Comment:
You need this in more places. Check out the other return code paths.

Or to make it easier, you might wrap the invocation of this entire function in another that inspects the error as necessary.

Reply:
Remove now?

Reply:
ended up adding a new function because of go's type/value interface nil idiosyncrasies